Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -172,7 +218,47 @@ private void GenerateCorrelationIdAndAddToDictionary(string ikey, string appId)
                 SdkInternalOperationsMonitor.Exit();
             }
         }
+#else
+        /// <summary>
+        /// Retrieves the app id given the instrumentation key.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key for which app id is to be retrieved.</param>
+        /// <returns>App id.</returns>
+        private Task<string> FetchAppIdFromService(string instrumentationKey)
+        {
+            var task = new Task<string>(() =>
```

Comment:
You never start this task. This will hang forever. Please fix this on develop. User TaskFactory.StartNew to create your task.